plugins/modules/host_errata_info.py

```diff
@@ -0,0 +1,103 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+# (c) 2021 Evgeni Golov
+#
+# This program is free software: you can redistribute it and/or modify
+# it under the terms of the GNU General Public License as published by
+# the Free Software Foundation, either version 3 of the License, or
+# (at your option) any later version.
+#
+# This program is distributed in the hope that it will be useful,
+# but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+# GNU General Public License for more details.
+#
+# You should have received a copy of the GNU General Public License
+# along with this program.  If not, see <http://www.gnu.org/licenses/>.
+
+from __future__ import absolute_import, division, print_function
+__metaclass__ = type
+
+
+DOCUMENTATION = '''
+---
+module: host_errata_info
+version_added: 2.1.0
+short_description: Fetch information about Host Errata
+description:
+  - Fetch information about Host Errata
+author:
+  - "Evgeni Golov (@evgeni)"
+options:
+  host:
+    description:
+      - Name of the host to fetch errata for.
+    required: true
+    type: str
+  content_view:
+    description:
+      - Calculate Applicable Errata based on a particular Content View.
+      - Required together with I(lifecycle_environment).
+    required: false
+    type: str
+  lifecycle_environment:
+    description:
+      - Calculate Applicable Errata based on a particular Lifecycle Environment.
+      - Required together with I(content_view).
+    required: false
+    type: str
+extends_documentation_fragment:
+  - theforeman.foreman.foreman
+  - theforeman.foreman.foreman.infomodulewithoutname
+'''
+
+EXAMPLES = '''
+- name: "Find errata for host"
+  theforeman.foreman.host_info:
+    username: "admin"
+    password: "changeme"
+    server_url: "https://foreman.example.com"
+    host: "host.example.com"
+'''
+
+RETURN = '''
+host_errata:
+  description: List of all found errata for the host and their details
+  returned: success
+  type: list
+  elements: dict
+'''
+
+from ansible_collections.theforeman.foreman.plugins.module_utils.foreman_helper import (
+    ForemanInfoAnsibleModule
+)
+
+
+class ForemanHostErrataInfo(ForemanInfoAnsibleModule):
+    pass
+
+
+def main():
+    module = ForemanHostErrataInfo(
+        foreman_spec=dict(
+            name=dict(invisible=True),
+            host=dict(type='entity', required=True),
+            content_view=dict(type='entity'),
+            environment=dict(invisible=True, type='entity', resource_type='lifecycle_environments', scope=['organization']),
+            lifecycle_environment=dict(),
```

Comment:
@mdellweg is there a better way to achieve this? the API expects a "environment_id", but I really prefer to call it "lifecycle_environment" in the params. also see the crude hack in L95+

Reply:
Isn't `flat_name` doing exacly that?

Don't we have the same confusing names elsewhere?

Reply:
Wald, Bäume.

Thanks. That was it. I *knew* we have that, and couldn't think of it.